GhidraPython/goresym_rename.py

```diff
@@ -0,0 +1,174 @@
+    if obj is None:
+        return False
+
+    try:
+        iter(obj)
+    except:
+        return False
+
+    return True
+
+def _entry_addr_estimator(hints):
+    if not iterable(hints["StdFunctions"]):
+        print("No StdFunctions present, required for entry addr estimation")
+        return None
+
+    # StdFunctions should have _rt0_[ARCH]_[OS](_lib)?, this maps to entry
+    candidates = [f for f in hints["StdFunctions"] if "_rt0_" in f["FullName"]]
+
+    if len(candidates) == 0:
+        print("No rt0 functions found")
+        return None
+
+    func = candidates[0]
+    if len(candidates) > 1:
+        # Just take the longest _rt0_ candidate
+        for c in candidates[1:]:
+            if len(c["FullName"]) > len(func["FullName"]):
+                func = c
+
+    for f in fm.getFunctions(True):
+        if f.name == "entry":
+            return f.entryPoint.offset-func["Start"]
+
+    return None
+
+def _pclntab_estimator(hints):
+    # Expected pclntab name per exec format
+    exec_fmt = currentProgram.getMetadata()["Executable Format"]
+    pclntabs = []
+    if exec_fmt == "Executable and Linking Format (ELF)":
+        # https://github.com/mandiant/GoReSym/blob/0c729523ed542f24b091e433204fbc6b02c88b31/objfile/elf.go#L89
+        pclntabs.append(".gopclntab")
+    elif exec_fmt == "Portable Executable (PE)":
+        # https://github.com/mandiant/GoReSym/blob/0c729523ed542f24b091e433204fbc6b02c88b31/objfile/pe.go#L130
+        pclntabs.append("runtime.pclntab")
+        pclntabs.append("runtime.epclntab")
+    elif exec_fmt == "Mac OS X Mach-O":
+        # https://github.com/mandiant/GoReSym/blob/0c729523ed542f24b091e433204fbc6b02c88b31/objfile/macho.go#L111
+        pclntabs.append("__gopclntab")
+    else:
+        # Try legacy
+        # https://github.com/mandiant/GoReSym/blob/0c729523ed542f24b091e433204fbc6b02c88b31/objfile/pe.go#L147
+        pclntabs.append("pclntab")
+        pclntabs.append("epclntab")
+        
+    # Check if pclntab is present
+    for p in pclntabs:
+        pclntab = currentProgram.memory.getBlock(p)
+        if pclntab is not None:
+            # With pclntab, offset should be a matter of TabMeta VA
+            tmva = hints["TabMeta"]["VA"]
+            text = currentProgram.memory.getBlock(".text")
+            return text.start.offset-tmva
+
+    return None
+
+def _func_map_estimator(hints):
+    offs = []
+    # Try to guess with known function names
+    grs_funcs = hints["UserFunctions"] + hints["StdFunctions"]
+    # Match function lists and collect possible offsets
+    for f in fm.getFunctions(True):
+        name = f.name
+        if "FUN_" in name or name == "entry":
+            continue
+        for gf in grs_funcs:
+            if name in gf["FullName"]:
+                offs.append(f.entryPoint.offset-gf["Start"])
+
+    winner, winner_count = None, 0
+    # Count given offsets, most counts is our estimate
+    for o in offs:
+        count = offs.count(o)
```

Comment:
another O(n**2) algorithm.

i think this can be replaced with a use of [`collections.Counter`](https://docs.python.org/2/library/collections.html#collections.Counter), maybe like:

```py
winner, winner_count = collections.Counter(offs).most_common[0]
```

Reply:
Agreed. Moved to using `collections.Counter`.